x-pack/plugin/security/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -7,6 +7,11 @@ grant {
   // needed because of SAML (cf. o.e.x.s.s.RestorableContextClassLoader)
   permission java.lang.RuntimePermission "getClassLoader";
   permission java.lang.RuntimePermission "setContextClassLoader";
+  // needed during initialization of OpenSAML library where xml security algorithms are registered
+  // see https://github.com/apache/santuario-java/blob/e79f1fe4192de73a975bc7246aee58ed0703343d/src/main/java/org/apache/xml/security/utils/JavaUtils.java#L205-L220
+  // and https://git.shibboleth.net/view/?p=java-opensaml.git;a=blob;f=opensaml-xmlsec-impl/src/main/java/org/opensaml/xmlsec/signature/impl/SignatureMarshaller.java;hb=db0eaa64210f0e32d359cd6c57bedd57902bf811#l52
+  // which uses it in the opensaml-xmlsec-impl
+  permission java.security.SecurityPermission "org.apache.xml.security.register";
```

Comment:
This [change](https://github.com/apache/santuario-java/commit/db7f9d065afaa1d8aa3a02b1e36001dffe996eb5) removed the `doPrivileged()` in xmlsec, which in turn means that every class in the whole execution thread needs the `java.security.SecurityPermission "org.apache.xml.security.register";` permission for registering the xml security algorithms. :(